src/hotspot/cpu/x86/stubGenerator_x86_64_poly_mont.cpp

```diff
@@ -564,7 +564,7 @@ address StubGenerator::generate_intpoly_montgomeryMult_P256() {
   address start = __ pc();
   __ enter();
 
-  if (EnableX86ECoreOpts && UseAVX > 1) {
+  if (VM_Version::supports_avxifma()) {
```

Comment:
As per the latest architecture-instruction-set-extensions-programming-reference document, the upcoming Diamond Rapdis CPU has AVX-IFMA but its also a P-Core Xeon with 512 bit flavour of IFMA, do you think it is performant to generate montgomeryMultiplyAVX2 for it ?

![image](https://github.com/user-attachments/assets/f1247484-180d-4a65-9b79-745a388c0552)

Reply:
Thanks for spotting this! I was wondering about this if statement being 'future-proof'.. I think its best to just flip the order of the if to prefer the AVX512 when available (it will make the diff look bigger, which is why I originally avoided it..)